oauthproxy.go

```diff
@@ -292,6 +293,10 @@ func (p *OAuthProxy) buildServeMux(proxyPrefix string) {
 	// Use the encoded path here so we can have the option to pass it on in the upstream mux.
 	// Otherwise something like /%2F/ would be redirected to / here already.
 	r := mux.NewRouter().UseEncodedPath()
+
+	// Trace all requests
+	r.Use(otelmux.Middleware("oauth2-proxy"))
```

Comment:
Looks like there's a lot of global state being used by this package, would prefer if we could create an instance rather than setting globals and sharing like this, any option for that?